cai_causal_graph/time_series_causal_graph.py

```diff
@@ -118,6 +118,7 @@ def __init__(
         self._summary_graph: Optional[CausalGraph] = None
         self._stationary_graph: Optional[TimeSeriesCausalGraph] = None
         self._minimal_graph: Optional[TimeSeriesCausalGraph] = None
+        self.lag_to_nodes, self.variable_name_to_nodes = self.cache_nodes()
```

Comment:
the way you have designed your code - you are using these properties **before** they are declared in the constructor - since these properties are used inside the `cache_nodes` method, which is wrong. Either declare them first, then use them, or make sure that `cache_nodes` does not use those properties.